### ISSUE TITLE
Cilium 1.9.1 does not work in AWS-CNI Chaining with aws-node 1.7.5 on EKS 1.18

### ISSUE THREAD

**Issue by kovalyukm** [cilium/cilium#14379]
## Bug report

Cilium 1.8.6 and 1.9.1 does not work on upgraded EKS cluster:
Kubernetes version | 1.18
Amazon VPC CNI plug-in | 1.7.5
DNS (CoreDNS) | 1.7.0
KubeProxy | 1.18.9

Installation and upgrading according to Cilium [doc](https://docs.cilium.io/en/v1.9/gettingstarted/cni-chaining-aws-cni/#setup-cluster-on-aws) 

CoreDNS pods and newly-created pods stuck in ContainerCreating with errors like:
```
Failed to create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "48fa873e8b20b0430e8d2f9717face03d920662474e5d9092b1b58400b2a71ad" network for pod "coredns-59458dc98-2275d": networkPlugin cni failed to set up pod "coredns-59458dc98-2275d_kube-system" network: invalid character '{' after top-level value

(combined from similar events): Failed to create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "4e631da4b0797f61b7a2c44f473cebff19c076b96444880f5e2eaea2ee023a85" network for pod "coredns-59458dc98-2275d": networkPlugin cni failed to set up pod "coredns-59458dc98-2275d_kube-system" network: netplugin failed but error parsing its diagnostic message "{\"level\":\"debug\",\"ts\":\"2020-12-11T17:48:25.839Z\",\"caller\":\"routed-eni-cni-plugin/cni.go:123\",\"msg\":\"MTU not set, defaulting to 9001\"}\n{\"level\":\"info\",\"ts\":\"2020-12-11T17:48:25.839Z\",\"caller\":\"routed-eni-cni-plugin/cni.go:117\",\"msg\":\"Received CNI add request: ContainerID(4e631da4b0797f61b7a2c44f473cebff19c076b96444880f5e2eaea2ee023a85) Netns(/proc/12483/ns/net) IfName(eth0) Args(IgnoreUnknown=1;K8S_POD_NAMESPACE=kube-system;K8S_POD_NAME=coredns-59458dc98-2275d;K8S_POD_INFRA_CONTAINER_ID=4e631da4b0797f61b7a2c44f473cebff19c076b96444880f5e2eaea2ee023a85) Path(/opt/cni/bin) argsStdinData({\\\"cniVersion\\\":\\\"0.3.1\\\",\\\"name\\\":\\\"aws-cni\\\",\\\"type\\\":\\\"aws-cni\\\",\\\"vethPrefix\\\":\\\"eni\\\"})\"}\n{\"level\":\"debug\",\"ts\":\"2020-12-11T17:48:25.839Z\",\"caller\":\"routed-eni-cni-plugin/cni.go:117\",\"msg\":\"MTU value set is 9001:\"}\n{\"level\":\"error\",\"ts\":\"2020-12-11T17:48:25.842Z\",\"caller\":\"routed-eni-cni-plugin/cni.go:117\",\"msg\":\"Failed to assign an IP address to container 4e631da4b0797f61b7a2c44f473cebff19c076b96444880f5e2eaea2ee023a85\"}\n{\n \"code\": 100,\n \"msg\": \"add cmd: failed to assign an IP address to container\"\n}": invalid character '{' after top-level value
```

Apparently the reason is:
ls /etc/cni/net.d/*
/etc/cni/net.d/05-cilium.conflist  /etc/cni/net.d/10-aws.conflist

Value of helm release fix errors above and pods are creating with assigning IP addresses (on newly-created nodes):
```yaml
    cni:
      customConf: true
```
BUT coredns pods are continuously restarted by cilium-operator:
```
2020-12-11T17:06:32.178716384Z level=info msg="Restarting unmanaged kube-dns pod kube-system/coredns-59458dc98-7cs4d started 30.178362095s ago" subsys=cilium-operator-generic
2020-12-11T17:06:32.195368360Z level=debug msg="Controller func execution time: 16.914056ms" name=restart-unmanaged-kube-dns subsys=controller uuid=1941cc5a-3bd2-11eb-bde4-02a1a7cad33b
2020-12-11T17:06:38.066759823Z level=debug msg="Controller func execution time: 2.492µs" name=update-cilium-nodes-pod-cidr subsys=controller uuid=18e4bd0e-3bd2-11eb-bde4-02a1a7cad33b
2020-12-11T17:06:41.926142257Z level=debug msg="Controller func execution time: 3.787623ms" name=k8s-heartbeat subsys=controller uuid=0d5eedd4-3bd2-11eb-bde4-02a1a7cad33b
2020-12-11T17:06:47.195554178Z level=debug msg="Found unmanaged kube-dns pod kube-system/coredns-59458dc98-v7z7p" subsys=cilium-operator-generic
2020-12-11T17:06:47.195581030Z level=info msg="Restarting unmanaged kube-dns pod kube-system/coredns-59458dc98-v7z7p started 30.195440502s ago" subsys=cilium-operator-generic
2020-12-11T17:06:47.206656218Z level=debug msg="Controller func execution time: 11.028395ms" name=restart-unmanaged-kube-dns subsys=controller uuid=1941cc5a-3bd2-11eb-bde4-02a1a7cad33b
2020-12-11T17:06:53.071588737Z level=debug msg="Controller func execution time: 2.142µs" name=update-cilium-nodes-pod-cidr subsys=controller uuid=18e4bd0e-3bd2-11eb-bde4-02a1a7cad33b
2020-12-11T17:07:02.207091524Z level=debug msg="Found unmanaged kube-dns pod kube-system/coredns-59458dc98-n88bk" subsys=cilium-operator-generic
2020-12-11T17:07:02.207138452Z level=debug msg="Found unmanaged kube-dns pod kube-system/coredns-59458dc98-72grx" subsys=cilium-operator-generic
```

Please, help fix this issue.

**General Information**

- Cilium version (run `cilium version`)
1.9.1

- Kernel version (run `uname -a`)
Linux 4.14.203-156.332.amzn2.x86_64 #1 SMP Fri Oct 30 19:19:33 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

- Orchestration system version in use (e.g. `kubectl version`, Mesos, ...)
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.4", GitCommit:"d360454c9bcd1634cf4cc52d1867af5491dc9c5f", GitTreeState:"clean", BuildDate:"2020-11-12T01:09:16Z", GoVersion:"go1.15.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.9-eks-d1db3c", GitCommit:"d1db3c46e55f95d6a7d3e5578689371318f95ff9", GitTreeState:"clean", BuildDate:"2020-10-20T22:18:07Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}

**How to reproduce the issue**

Install Cilium 1.9.1 in AWS-CNI Chaining mode on EKS 1.18 with Amazon VPC CNI plug-in 1.7.5 using Cilium chart as described in Cilium doc.

**Comment by joestringer** [comment]
Hi @kovalyukm , thanks for the report. This issue is caused by the upstream aws-cni plugin, as reported here: https://github.com/aws/amazon-vpc-cni-k8s/issues/1314 . It looks like the fix has been merged and they are planning to release a new version of the aws-cni plugin to address this issue (likely to be v1.7.8). In the mean time, it looks like there is a mitigation posted on the linked issue above.

**Comment by springroll12** [comment]
This fix for this has been delayed until at least April 2021. Are there any workarounds known?

**Comment by joestringer** [comment]
This was addressed in the Feb 18 release v1.7.9 for aws-cni: https://github.com/aws/amazon-vpc-cni-k8s/releases/tag/v1.7.9.
